File: src/asignaciones_y_variables_06.c
/* 
*
*	Título: EJERCICIOS DE VARIABLES Y ASIGNACIONES 6
*	Consigna: Ingresar los lados de un rectángulo y calcular su diagonal principal, superficie y perímetro.
*	Alumno: Nicolás Esteban Olmedo
*
*/

/* Directivas del preprocesador */
#include <stdio.h>
#include <math.h>

/* Macros del procesador */

/* Definiciones globales */
	/* Definición de variables globales */

	/* Definición de prototipos de funciones */
void clrscr(void);
float calculo_hip(float arg0, float arg1);
float calculo_perim(float arg0, float arg1);
float calculo_area(float arg0, float arg1);

/* Función principal */
int main() {
	clrscr();
	/* Definición de variables locales */
	float cateto0, cateto1, hipotenusa, perimetro, area;
	/* Input de data */
	printf("Insertar cateto menor: ");
	scanf("%f", &cateto0);
	printf("Insertar cateto mayor: ");
	scanf("%f", &cateto1);
	
	/* Procesamiento de data */
	hipotenusa = calculo_hip(cateto0, cateto1);
	perimetro = calculo_perim(cateto0, cateto1);
	area = calculo_area(cateto0, cateto1);
	
	/* Output de data */
	printf("Diagonal principal = %f\n Perímetro = %f\n Área = %f\n", hipotenusa, perimetro, area);
		return 0;
}
/* Otras funciones */
	/* Definición clrscr() */
void clrscr() {
	system("clear");
}
	/* Definición calculo_hip() */
float calculo_hip(float arg0, float arg1) {
	float res;
	res = sqrt(pow(arg0,2) + pow(arg1,2));
}

	/* Definición calculo_perim() */
float calculo_perim(float arg0, float arg1) {
	float res;
	res = arg0*2 + arg1*2;
	return res;
}

	/* Definición calculo_area() */
float calculo_area(float arg0, float arg1) {
	float res;
	res = arg0 * arg1;
	return res;
}

File: C#files/C#practice/src/asignaciones_y_variables_06.cs
/*
 * Ingresar los lados de un rectángulo y calcular su diagonal principal, superficie y perímetro.
 */

using System;

namespace AsignacionesVariables06 {
	class Program {
		static void Main(string[] args) {
			float ladoA, ladoB, hipotenusa, superficie, perimetro;
			string tempVal;

			//Input de data
			Console.Write("Ingresar base: ");
			tempVal = Console.ReadLine();
			ladoA = Convert.ToSingle(tempVal);

			Console.Write("Ingresar altura: ");
			tempVal = Console.ReadLine();
			ladoB = Convert.ToSingle(tempVal);

			//Procesamiento de data
			hipotenusa = CalculoHipotenusa(ladoA, ladoB);
			perimetro = CalculoPerimetro(ladoA, ladoB, hipotenusa);
			superficie = CalculoSuperficie(ladoA, ladoB);

			//Output de data
			Console.WriteLine("Diagonal principal = " + hipotenusa);
			Console.WriteLine("Perimetro = " + perimetro);
			Console.WriteLine("Superficie = " + superficie);
		}

		private static float CalculoHipotenusa(float a, float b) {
			float tempRes = (float)Math.Sqrt( (float)Math.Pow(a,2) + (float)Math.Pow(b,2));
			return tempRes;
		}

		private static float CalculoPerimetro(float a, float b, float c) {
			float tempRes = a + b + c;
			return tempRes;
		}

		private static float CalculoSuperficie(float a, float b) {
			float tempRes = a*b;
			return tempRes;
		}
	}
}
